doc/developer-notes.md

```diff
@@ -1004,10 +1004,6 @@ Strings and formatting
     buffer overflows, and surprises with `\0` characters. Also, some C string manipulations
     tend to act differently depending on platform, or even the user locale.
 
-- For `strprintf`, `LogInfo`, `LogDebug`, etc formatting characters don't need size specifiers.
-
-  - *Rationale*: Bitcoin Core uses tinyformat, which is type safe. Leave them out to avoid confusion.
```

Comment:
Although, the phrase "size specifiers" is used here which I believe is incorrect, the spirit of this remark is still helpful. 

Using the terminology from this article: https://en.cppreference.com/w/c/io/fprintf I believe the spirit of this note is, "- For `strprintf`, `LogInfo`, `LogDebug`, etc: When using conversion specifiers like `%d`, don't use length modifiers (e.g.: `%ld`).

Reply:
The reason i added this back in the day is that 64-bit types are awful to handle in a cross-platform way in `printf` family functions. This necessitated use of macros like `PRIx64`, which didn't even work on windows (or something) and actually actively break with tinyformat. So just being able to use `%d` and `%x` is very nice and saves thinking about this.

Reply:
It's not obvious (to me), but as long as the compiler tells me what to do... otherwise a note is helpful.

Reply:
Well the idea was that by the time the compiler gets to it, a developer may already have done a frustrating search "what % expression do i need to use". Documenting it could avoid that (i don't think this particular blurb is "stale" in any way besides "not changed in a long time). 

That said, there are tons of examples of `strprintf` in the code, and nothing uses length modifiers. Tinyformat also uses `%s` for strings without `c_str()`, which isn't explicitly documented.